### PR TITLE
Add release-branch-jobs 1.25

### PR DESCRIPTION
**Pull request by MaxHorstmann**
#7 added 1.24

this adds 1.25

1.25.yaml is based on https://github.com/kubernetes/test-infra/blob/a447b9e79b897a029b5809c3ea5b4f133d5c8452/config/jobs/kubernetes/sig-release/release-branch-jobs/1.25.yaml (now [deleted](https://github.com/kubernetes/test-infra/commit/7714bb3296dbb18e4ca9e18ce3e2c29429128e02) because 1.25 no longer supported)

Test PR: https://github.com/aks-lts/kubernetes/pull/16

### CODE REVIEW

File: config/prow/release-branch-jobs/1.25.yaml [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+              - name: SKIP
+                value: \[Slow\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]|PodSecurityPolicy|LoadBalancer|load.balancer|Simple.pod.should.support.exec.through.an.HTTP.proxy|subPath.should.support.existing|NFS|nfs|inline.execution.and.attach|should.be.rejected.when.no.endpoints.exist
+              - name: PARALLEL
+                value: "true"
+            image: gcr.io/k8s-staging-test-infra/krte:v20231012-0288f8bc6c-1.25
+            name: ""
+            resources:
+              limits:
+                cpu: "7"
+                memory: 9000Mi
```
Comment (haitch) [inline config/prow/release-branch-jobs/1.25.yaml:278]: 9Gi?

File: config/prow/release-branch-jobs/1.25.yaml [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+      path_alias: k8s.io/kubernetes
+      spec:
+        containers:
+          - args:
+              - kubetest2
+              - noop
+              - --test=node
+              - --
+              - --repo-root=.
+              - --gcp-zone=us-west1-b
```
Comment (haitch) [inline config/prow/release-branch-jobs/1.25.yaml:70]: do we need this?

File: config/prow/release-branch-jobs/1.25.yaml [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+              - runner.sh
+            env:
+              - name: WHAT
+                value: external-dependencies-version vendor vendor-licenses
+            image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20231113-7213ea5323-1.25
+            name: main
+            resources:
+              limits:
+                cpu: "2"
+                memory: 1288490188800m
```
[the remark this answers is not in the document]
Comment (MaxHorstmann) [reply config/prow/release-branch-jobs/1.25.yaml:151]: it is weird, [source](https://github.com/kubernetes/test-infra/blob/a447b9e79b897a029b5809c3ea5b4f133d5c8452/config/jobs/kubernetes/sig-release/release-branch-jobs/1.25.yaml)

File: config/prow/release-branch-jobs/1.25.yaml (hunk 1 above)
Comment (MaxHorstmann) [reply config/prow/release-branch-jobs/1.25.yaml:278]: yeah, that's what they use on k/k test infra.

I'll change it to 

```
            resources:
              limits:
                cpu: "4"
                memory: 8Gi
              requests:
                cpu: "2"
                memory: 4Gi
```

everywhere

File: config/prow/release-branch-jobs/1.25.yaml (hunk 2 above)
Comment (MaxHorstmann) [reply config/prow/release-branch-jobs/1.25.yaml]: seems GCP specific, let's remove it

**Comment by MaxHorstmann** [comment]
@haitch ok to merge?

**Review by haitch — APPROVED** [review]

**Pull request merged**